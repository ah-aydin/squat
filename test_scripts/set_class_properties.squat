class Person {
    int age;
    int height;
    string name;
}

Person global_guy = Person(9, 123, "global");

func main() {
    Person person = Person(29, 120, "short guy");
    int a = 0;
    int b = 1;

    a = b = person.age;
    println(a);
    println(b);

    println("Before change");
    println(person);
    
    person.age = 39;
    person.height = 199;
    person.name = "now dis guy is tall guy";

    println("After change");
    println(person);

    person.age = a = b = 320;
    println(person.age);
    println(a);
    println(b);
    println(person);

    person.name = "new name";
    println(person);

    println(global_guy);
    global_guy.name = "ofya";
    println(global_guy);
    person.name = global_guy.name;
    println(person);
    person.name = global_guy.name = "last name";
    println(person);
    println(global_guy);
}

