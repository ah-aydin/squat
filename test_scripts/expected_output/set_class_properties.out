Compiling and running file: test_scripts/set_class_properties.squat
29
29
Before change
<instance of Person [Int(29), Int(120), String("short guy")]>
After change
<instance of Person [Int(39), Int(199), String("now dis guy is tall guy")]>
320
320
320
<instance of Person [Int(320), Int(199), String("now dis guy is tall guy")]>
<instance of Person [Int(320), Int(199), String("new name")]>
<instance of Person [Int(9), Int(123), String("global")]>
<instance of Person [Int(9), Int(123), String("ofya")]>
<instance of Person [Int(320), Int(199), String("ofya")]>
<instance of Person [Int(320), Int(199), String("last name")]>
<instance of Person [Int(9), Int(123), String("last name")]>
